specifyweb/frontend/js_src/lib/components/QueryBuilder/ToMap.tsx

```diff
@@ -209,26 +223,33 @@ export function QueryToMapDialog({
         ...localityData.current,
         ...extractLocalities(results, localityMappings),
       ];
-      setInitialData((initialData) =>
-        typeof initialData === 'object'
-          ? initialData
-          : {
-              localityData: localityData.current.map(
-                ({ localityData }) => localityData
-              ),
-              onClick: createClickCallback(tableName, localityData.current),
-            }
-      );
       markerEvents.trigger('updated');
     },
     [tableName, localityMappings, markerEvents]
   );
 
-  // Add initial results
-  React.useEffect(() => handleAddPoints(results), [handleAddPoints]);
   useFetchLoop(handleFetchMore, handleAddPoints);
 
-  React.useEffect(() => {
+  /*
+   * The below is used for sanity checking at un-mount.
+   * A unit test for this functionality is tricky. A runtime check is simpler
+   */
+  const mapRef = React.useRef(map);
+  mapRef.current = map;
+
+  React.useEffect(
+    () => () => {
+      if (mapRef.current === undefined) return;
+      if (mapRef.current?.sp7MarkerCount !== markerCountRef.current) {
+        console.warn(
+          `Expected the counts to match: Expected: ${markerCountRef.current}. Got: ${mapRef.current?.sp7MarkerCount}`
+        );
+      }
+    },
```

Comment:
This is a runtime check. Done as part of the destructor. Useful to debug this in the feature in the future. This test will fail in production because the `sp7MarkerCount` will be greater than the `markerCountRef.current`.

We could crash here, but that seems to severe.